service/service.go

```diff
@@ -184,6 +194,7 @@ func (s *Service) Boot() {
 		s.Alerter.StartAlerts()
 
 		// Start the framework.
-		s.CustomObjectFramework.Boot()
+		go s.CRDFramework.Boot()
```

Comment:
This probably comes from operatorkit but these goroutines will likely leak, could we use a context here to cleanly stop them when stopping the operator?

Reply:
These are the main processes. We only call boot once. Nothing is leaked. What we need to do though is to add some graceful shutdown but IMO this is a bigger thing because the design is not clear to me personally.

Reply:
My understanding is the `sync.Once` means we only call the framework Boot method once. The framework is responsible for ensuring we stop cleanly and don't leak goroutines.

https://github.com/giantswarm/operatorkit/blob/25a4a3648e62fcb141d02bf1aba9fd905a94f0ea/framework/framework.go#L204

Reply:
Thanks for your responses! AIUI when we stop the operator we are stopping the main goroutine, the goroutine initiated in that lines and enters this loop https://github.com/giantswarm/operatorkit/blob/25a4a3648e62fcb141d02bf1aba9fd905a94f0ea/framework/framework.go#L480 would leak in that case, pls correct me if I'm wrong.

IMO it would be very easy to pass a context to those methods (instead of instantiating it here https://github.com/giantswarm/operatorkit/blob/25a4a3648e62fcb141d02bf1aba9fd905a94f0ea/framework/framework.go#L205) and wait for a stop signal, canceling the context in case it is received. I think that this way we would have the graceful shutdown Tim mentions, WDYT? Anyway I need to dig deeper to fully understand operatorkit internals.

Reply:
Technically you are right that the goroutine leaks, but in reality it does not matter because it happens once and only on process shutdown. Lets come up with a nice idea about how to shutdown our microkit services and then we can execute a plan once we have it. I don't want to hurry this.